Volnychina

Elena-voln opened the pull request:
Без sky veiw, но траекторию видно

Elena-voln committed:
программа расчета, очень сырая

Elena-voln committed:
Предварительно: работает, орбиту спутника видно.

Без учета небесных тел. Почему то луна и солнце вносят дичайшие ошибки (можно потестить в файле F все еще есть коды)

Elena-voln committed:
Осталось только SKY Veiw

Elena-voln committed:
Отчет без sky veiw

Elena-voln committed:
просто не пуллится без этого

Elena-voln committed:
Merge remote-tracking branch 'origin/volnychina' into volnychina

Korogodin commented:
Привет, Лена! Я верю, что ты можешь лучше)

SkyView - это такой же график траектории спутника, только в сферических координатах (азимут и угол места, третью координату, расстояние, не отображаем). И из системы координат, связанной с приемником. 

Соответственно, шаг первый: сделать преобразование координат в точку, где стоит приемник. Иначе говоря из ECEF в ENU. В матлабе под это даже готовая функция есть, но можно и свою сделать.

Шаг второй: перейти к сферическим координатам. Опять же задачка то ли из школы, то ли со второго курса. 

Шаг третий: построить результат. Для отображения в полярных координатах в матлабе есть готовый polarplot

Кроме того, ты всегда можешь вдохновляться коллегами.

Elena-voln committed:
пытаюсь вылечить время

Elena-voln committed:
Два скачка координат. Почему- не ясно. Явно из-за времени

Elena-voln committed:
Вся траектория правильно отображается, без косяков

Урааа!!!!

Elena-voln committed:
траектория вырисовывается, но skyveiw еще не верный

почему- не понимаю

Korogodin commented:
Привет, Лена! Ты тут уже месяц не коммитила, от этого грустно...а давно пора выставлять КМ

Elena-voln committed:
Merge tag 'HEAD' into volnychina

Elena-voln committed:
просто схожу с ума от гиткракена

Elena-voln committed:
Скайвью не совпадает с действительностью

Все траектории выводятся, орбиты строятся, но именно скайвью почему то нет. Не получается найти ошибку

Elena-voln committed:
Merge remote-tracking branch 'origin/volnychina' into volnychina

Elena-voln committed:
Merge remote-tracking branch 'origin/volnychina' into volnychina

Korogodin commented:
Вот тут у тебя описка:
> %Coordinates transformation to an inertial reference frame:
> %Position

А потом ты зачем-то вращаешь несколько раз матрицу, которая у тебя получается после решения дифура. В итоге доходишь до того, что у тебя в ней порядок точек инвертирован, спутник "летит" в противоположную сторону

Elena-voln commented:
![image](https://user-images.githubusercontent.com/57173818/82305942-0afb3b80-99c7-11ea-9968-6b11914c8294.png)
Я поворачивала векторы измерений находящихся в промежутке от te до start
на рисунке мое предположение почему так надо делать

Korogodin commented:
Так или иначе, у тебя в массиве время идет в обратную сторону

Elena-voln commented:
Не поняла почему- вывела от момента "стыка" массив времени, там все нормально
![image](https://user-images.githubusercontent.com/57173818/82310139-af33b100-99cc-11ea-830d-0084c562f788.png)
Попробовала инвертировать время,  скайвью не сошелся :(
![image](https://user-images.githubusercontent.com/57173818/82310217-c5da0800-99cc-11ea-84e7-0718f82b8d1f.png)
А вот траектория стала выглядеть приличней
![image](https://user-images.githubusercontent.com/57173818/82310276-d68a7e00-99cc-11ea-8198-fab32e92be3b.png)

Elena-voln commented:
К предыдущему сообщению
te=60300  там выведено от te-10 до te+10

Korogodin commented:
![image](https://user-images.githubusercontent.com/747285/82310587-44cf4080-99cd-11ea-8bb6-0fe7e3c6fc1f.png)

Korogodin commented:
Лена, как твои успехи?

Elena-voln committed:
время теперь не обратно

Elena-voln commented:
Я еще раз прошерстила весь код, но даже с исправленным временем-не то.  Я совсем не понимаю, что не так

![image](https://user-images.githubusercontent.com/57173818/82876565-aa797a80-9f41-11ea-8ede-573f14fa88b1.png)

Оно даже по форме не похоже на то что надо
![image](https://user-images.githubusercontent.com/57173818/82876655-cf6ded80-9f41-11ea-9458-7a64fdb27479.png)

Korogodin commented:
Ты не закоммитила, мне нечего смотреть

Elena-voln commented:
Я пуш сделала, Потрикеева тоже комит видит

![image](https://user-images.githubusercontent.com/57173818/82878584-56bc6080-9f44-11ea-86be-65439cf5f0f2.png)
и приложение видит что все в сеть передано
![image](https://user-images.githubusercontent.com/57173818/82878670-75baf280-9f44-11ea-8c56-9755a3354643.png)

Elena-voln commented:
могу еще один ПР сделать

Korogodin commented:
Я начинаю догадываться, в чем наша проблема. Ты коммитишь и в ветку volnychina, и в ветку master, но ветка master у тебя отстает. Я же не обратил на это внимание и смотрел master

Korogodin commented:
Первый участок, где до te, у тебя состоит из NaN. 
Второй считается с точностью в пару км, хотя бы так. 
А дальше ты зачем-то опять занимаешься своим любимым делом - переворачиваешь. 
Теперь Землю вверх ногами перевернула

![image](https://user-images.githubusercontent.com/747285/82889918-1fee4680-9f54-11ea-9eab-73b85a1b5ff8.png)

Elena-voln committed:
точка возврата. Все работает, без луны и солнца

Elena-voln committed:
Готово. Отчет в PDF файле

Elena-voln commented:
Все готово!

Korogodin commented:
Формально отчет получился не по ГОСТ, но давай закроем на это глаза, ты тренируешься в LaTeX. Для гостовского оформления надо воспользоваться соответствующим стилем:
https://github.com/latex-g7-32/latex-g7-32

Этап 2 принят. 
Третьего числа у нас защита онлайн, ждать ли что-нибудь по третьему этапу? Надо уже давно КМ проставить, боюсь, что со следующей недели ввод будет недоступен
